### ISSUE TITLE
Elevation

### ISSUE THREAD

**Issue by ssbittle** [prenda-school/prenda-spark#186]
### Design Change: 

Updated Elevation styling.

- **E100**
- **E200**
- **E300**
- **E400**
- **E500**

Reference found [here](https://www.figma.com/file/w9lodWeivyS4GHdBNIYaJr/?node-id=2%3A154).

**Comment by WilliamKelley** [comment]
### Design feedback
Since shadows are part of the global theme, changing things like how many options of elevation there are will have rippling repercussions across our components and any consumer use of those values independently (i.e. when a product design requires them to pull in an elevation 1 time). Such a change is _breaking_, because completing removing something like `elevation[9]` means old code will break the application and stop it from running.

Given that, how soon can we expect elevation values to be considered "final"? This conversation extends to all Foundation values really. I'm not looking to force restrictions here, but it should definitely be on design's timeline because ever-changing values will mean we can never release a stable version of spark, or we release many major versions -- both are bad for our consuming projects.

### Before vs After

![image](https://user-images.githubusercontent.com/25781782/129226460-d8e4fb09-28c3-4b6a-b43f-d009de9cfee2.png)
![image](https://user-images.githubusercontent.com/25781782/129226474-db0e1f12-8209-4a5c-89ab-5894185516a7.png)

#### Implementation
Notes: 
  - always two shadows
  - first shadow is constant and appears like a light border
  - second shadow grows in `y-offset` and `blur-radius` and decreases it's transparency over the first three elevations; this type of shadow is a "key light shadow"

Given the above, our current shadow

```css
/* #072E44 or `neutral.navy` / `blue[5]` <=> rgb(7, 46, 68) */

/* E100 */
box-shadow: 0px 0px 1px 0px rgba(7, 46, 68, 0.31); /* #072E444F */
box-shadow: 0px 1px 1px 0px rgba(7, 46, 68, 0.24); /* #072E443D */

/* E200 */
box-shadow: 0px 0px 1px 0px rgba(7, 46, 68, 0.31); /* #072E444F */
box-shadow: 0px 3px 5px 0px rgba(7, 46, 68, 0.20); /* #072E4433 */

/* E300 */
box-shadow: 0px 0px 1px 0px rgba(7, 46, 68, 0.31); /* #072E444F */
box-shadow: 0px 8px 12px 0px rgba(7, 46, 68, 0.15); /* #072E4426 */

/* E400 */
box-shadow: 0px 0px 1px 0px rgba(7, 46, 68, 0.31); /* #072E444F */
box-shadow: 0px 10px 18px 0px rgba(7, 46, 68, 0.15); /* #072E4426 */

/* E500 */
box-shadow: 0px 0px 1px 0px rgba(7, 46, 68, 0.31); /* #072E444F */
box-shadow: 0px 18px 28px 0px rgba(7, 46, 68, 0.15); /* #072E4426 */
```

**Comment by WilliamKelley** [comment]
### Implementation



Given the above implementation notes, our current shadow creation function will have to be ditched. We can create a new abstraction function for the second shadow.

I'm not sure how to handle the fact that 
- we have only 5 shadows compared to Mui's 25 shadows
- some components underlying reliance or accessing to the default amount of shadows
- the terming of "E100" "E200", etc. -- should we make shadows an object with keys of `100`, `200`, etc.? Or simplify to 1, 2, etc. to keep the current array.

@seigler Since you did the initial shadows work, do you have any thoughts/feedback?

**Comment by WilliamKelley** [comment]
### Implementation

The currently implemented Spark Web components affected by elevation change are `Menu` and `Card`. They will have to be audited with this update.

**Comment by ssbittle** [comment]
@WilliamKelley Thank you for bringing this to my attention. You make a very very valid argument. 

In short, as a team we feel much more confident in this collection of shadows/elevation levels than our previous collection. 

I understand the repercussions updating core foundations can/will cause a hiccup. I do not foresee any major changes in near future that would impact foundations, outside of maybe colors.

If a case comes up, then design will be sure to bring that discussion to development to make a final decision.

We can consider this a final deliverable. 

I hope that insight helps.

**Comment by WilliamKelley** [comment]
Definitely! Thanks Spencer

**Comment by seigler** [comment]
IMO the benefit of using "E100" is that it leaves open the possibility of future finer control of elevation. If we adopt a numerical 1-5 it is simpler, but also more of a commitment to this level of resolution.

**Comment by WilliamKelley** [comment]
After reviewing `createTheme` source, I think we can get away with making it an object and adding 100, 200, etc. keys but we have to preserve defaults (1-25) for under-the-hood usages. But, I am unsure if we can make it work with TypeScript...

**Comment by WilliamKelley** [comment]
I don't think I'll be able to override it since Mui's `Shadows` a `type` and not `interface`

**Comment by WilliamKelley** [comment]
Closed by #188 

I've also updated the Figma file with the JS accessor equivalent

![image](https://user-images.githubusercontent.com/25781782/129403715-9da938e7-9e4d-46e4-af63-ea316f4cfb7b.png)
